lib/Solve.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+const crypto = require('crypto');
+const path = require('path');
+const fs = require('fs-extra');
+const spawn = require('cross-spawn');
+
+function sha1(string) {
+  return crypto.createHash('sha1').update(string).digest('hex');
+}
+
+function getDependenciesCached(
+  generatedCodeDir /*: string */,
+  elmJsonPath /*: string */,
+  projectElmJson /*: any */
+) /*: { direct: { [string]: string }, indirect: { [string]: string } } */ {
+  const hash = sha1(
+    JSON.stringify({
+      dependencies: projectElmJson.dependencies,
+      'test-dependencies': projectElmJson['test-dependencies'],
+    })
+  );
+
+  const cacheFile = path.join(generatedCodeDir, `dependencies.${hash}.json`);
```

Comment:
I really like this! 

I only wonder if there is a way of cleaning up old cache files so that we don't use up too much disk space.

Reply:
Do we need to worry about that?

- The files are very small (< 4 kB).
- elm.json changes very seldomly so there won’t be many versions.
- We don’t clean up anything else: I currently have 63 MainXXXXXXX.elm generated by elm-test, and also old generated code for previous versions of elm-test. This takes ~2 MB of space (not counting the nested elm-stuff folders).

Reply:
Wow, okay. That is a lot of MainXxxxxx.elm! I think we are probably okay then.